salt/modules/debian_ip.py

```diff
@@ -1498,6 +1498,13 @@ def _write_file_routes(iface, data, folder, pattern):
     '''
     Writes a file to disk
     '''
+    # Since the entire directory is executed with matching pattern
+    # ^[a-zA-Z0-9_-]+$ not all filenames are legitimate.
+    # In order to make routes work for vlan interfaces (e.g. bond0.12)
+    # the dots in the iface name will be replaced by underscores, so it
+    # will be properly executed
```

Comment:
because a route file for bond0.12 will be stored by default in /etc/network/if-up.d/route-bond0.12 (which is not executed, because it does not match the pattern)